colorfield/widgets.py

```diff
@@ -29,8 +33,17 @@ def get_context(self, name, value, attrs=None):
                 "value": value,
             }
         )
-        if "format" not in context:
-            context.update({"format": "hex"})
+        coloris = {
+            "format": context.get("format", "hex"),
+            "required": context.get("required", False),
+            "clearButton": not bool(context.get("required")),
+            "alpha": bool(context.get("alpha")),
+            "forceAlpha": bool(context.get("alpha")),
+            "swatches": context.get("swatches", []),
+            "swatchesOnly": bool(context.get("swatches", []))
+            and bool(context.get("swatches_only")),
+        }
+        context.update({"data_coloris": json.dumps(coloris)})
```

Comment:
No more need to convert data to json here (sorry).

Reply:
I tried your suggestion and I could not make it work fine.

Reply:
Let's keep it as it is.